hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/process/ConsumerProcessSupervisor.java

```diff
@@ -155,61 +155,63 @@ private void onConsumerProcess(Signal signal, java.util.function.Consumer<Consum
         }
     }
 
-    private void kill(SubscriptionName subscriptionName) {
-        if (!runningProcesses.hasProcess(subscriptionName)) {
-            logger.info("Process for subscription {} no longer exists", subscriptionName);
+    private void kill(Signal signal) {
+        if (!runningProcesses.hasProcess(signal.getTarget())) {
+            logger.info("Process for subscription {} no longer exists. Signal id {}", signal.getTarget(), signal.getId());
         } else {
-            logger.info("Interrupting consumer process for subscription {}", subscriptionName);
-            Future task = runningProcesses.getExecutionHandle(subscriptionName);
+            logger.info("Interrupting consumer process for subscription {}. Signal id {}", signal.getTarget(), signal.getId());
+            Future task = runningProcesses.getExecutionHandle(signal.getTarget());
             if (!task.isDone()) {
                 if (task.cancel(true)) {
-                    logger.info("Interrupted consumer process {}", subscriptionName);
+                    logger.info("Interrupted consumer process {}", signal.getTarget());
                 } else {
                     logger.error("Failed to interrupt consumer process {}, possible stale consumer",
-                            subscriptionName);
+                            signal.getTarget());
                 }
             } else {
-                runningProcesses.remove(subscriptionName);
-                logger.info("Consumer was already dead process {}", subscriptionName);
+                runningProcesses.remove(signal.getTarget());
+                logger.info("Consumer was already dead process {}", signal.getTarget());
             }
         }
     }
 
     private boolean isHealthy(ConsumerProcess consumerProcess) {
-        long delta = clock.millis() - consumerProcess.healthcheckRefreshTime();
-        if (delta > unhealthyAfter) {
-            logger.info("Lost contact with consumer {}, last seen {}ms ago", consumerProcess, delta);
-            return false;
-        }
-        return true;
+        return unhealthyAfter > consumerProcess.lastSeen();
```

Comment:
There is no test for this significant change of logic (please do add some tests), yet the variable (misspelled) healtcheckRefreshTime does not get updated. Or does it?